internal/kv/tikv/txn_tikv.go

```diff
@@ -698,3 +729,43 @@ func (kv *txnTiKV) removeTiKVMeta(ctx context.Context, key string) error {
 func (kv *txnTiKV) CompareVersionAndSwap(key string, version int64, target string) (bool, error) {
 	return false, fmt.Errorf("Unimplemented! CompareVersionAndSwap is under deprecation")
 }
+
+// CheckElapseAndWarn checks the elapsed time and warns if it is too long.
+func CheckElapseAndWarn(start time.Time, message string, fields ...zap.Field) bool {
+	elapsed := time.Since(start)
+	if elapsed.Milliseconds() > 2000 {
+		log.Warn(message, append([]zap.Field{zap.String("time spent", elapsed.String())}, fields...)...)
+		return true
+	}
+	return false
+}
+
+// Since TiKV cannot store empty key values, we assign them a placeholder held by EmptyValue.
+// Upon loading, we need to check if the returned value is the placeholder.
+func isEmptyByte(value []byte) bool {
+	if bytes.Equal(value, EmptyValueByte) {
+		return true
+	}
+	return false
+}
+
+// Return an empty string if the value is the Empty placeholder, else return actual string value.
+func convertEmptyByteToString(value []byte) string {
+	if isEmptyByte(value) {
+		return ""
+	} else {
+		return string(value)
+	}
+}
+
+// Convert string into EmptyValue if empty else cast to []byte. Will throw error if value is equal
+// to the EmptyValueString.
+func convertEmptyStringToByte(value string) ([]byte, error) {
+	if len(value) == 0 {
+		return EmptyValueByte, nil
+	} else if value == EmptyValueString {
+		return nil, fmt.Errorf("Value for key is reserved by EmptyValue: %s", EmptyValueString)
```

Comment:
Not sure if the caller will log it. it's probably better to also add a log.error here.